WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -127,9 +127,10 @@ extension CardPresentPaymentEventDetails {
             return .message(.tapSwipeOrInsertCard(
                 viewModel: PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageViewModel(
                     inputMethods: inputMethods)))
-
         case .paymentSuccess:
             return .message(.paymentSuccess(viewModel: PointOfSaleCardPresentPaymentSuccessMessageViewModel()))
+        case .paymentError(error: CollectOrderPaymentUseCaseError.couldNotRefreshOrder(let error), let tryAgain, _):
```

Comment:
@joshheald, this is where I need your opinion.

For now, order validation is one of the payment errors. Theoretically, I could introduce `validatingOrderError` at any stage. 

I made a decision to only introduce `validatingOrderError` when translating `CardPresentPaymentEvent` into `PointOfSaleCardPresentPaymentEventPresentationStyle`. However, you may have a more informed view on this.

Thanks!

Reply:
I agree with this approach

This order validation is part of the payment process, and it's in response to a `CardPresentPaymentEvent` produced as one of the first steps after starting a payment.

This happens after refreshing the order to check it's still valid for payment. IMO the POS doesn't really need to do that, because the order is only just created and won't be accessible later, so should be valid. However, it's part of the underlying payment code that we reuse, so it's better to handle it than try to prevent it from happening.

We may in future have other order validation errors, when they're added they should be handled elsewhere.